# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSet.h
# Commit: 43764851d067
# Full Hash: 43764851d067746870a65001786502244d7cf896
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-04-07 03:19:44
# Regressor Bug: 1694123
# File Overlap Count: 1
# Description:
#   Bug 1694123 - Pass userfont source index to font loader and decoder tasks, to avoid potential race with the field in the gfxUserFontEntry being reset on the main thread. r=layout-reviewers,emilio
#   
#   This is a preliminary step that is needed in order for the following patch to be safe. (It's
#   mostly just plumbing, although the patch looks big because we need to pass the index through
#   so many functions.)
# ==============================================================================

diff -r 9b111c26978a -r 43764851d067 layout/style/FontFaceSet.h
--- a/layout/style/FontFaceSet.h	Tue Apr 06 16:20:47 2021 +0000
+++ b/layout/style/FontFaceSet.h	Tue Apr 06 16:23:17 2021 +0000
@@ -72,7 +72,7 @@
         nsTArray<nsCOMPtr<nsIRunnable>>& aViolations) override;
 
     virtual nsresult StartLoad(gfxUserFontEntry* aUserFontEntry,
-                               const gfxFontFaceSrc* aFontFaceSrc) override;
+                               uint32_t aSrcIndex) override;
 
     void RecordFontLoadDone(uint32_t aFontSize, TimeStamp aDoneTime) override;
 
@@ -87,7 +87,7 @@
                                       uint8_t*& aBuffer,
                                       uint32_t& aBufferLength) override;
     virtual nsresult LogMessage(gfxUserFontEntry* aUserFontEntry,
-                                const char* aMessage,
+                                uint32_t aSrcIndex, const char* aMessage,
                                 uint32_t aFlags = nsIScriptError::errorFlag,
                                 nsresult aStatus = NS_OK) override;
     virtual void DoRebuildUserFontSet() override;
@@ -98,8 +98,8 @@
         const nsTArray<gfxFontVariation>& aVariationSettings,
         uint32_t aLanguageOverride, gfxCharacterMap* aUnicodeRanges,
         StyleFontDisplay aFontDisplay, RangeFlags aRangeFlags,
-        float aAscentOverride, float aDescentOverride,
-        float aLineGapOverride, float aSizeAdjust) override;
+        float aAscentOverride, float aDescentOverride, float aLineGapOverride,
+        float aSizeAdjust) override;
 
    private:
     RefPtr<FontFaceSet> mFontFaceSet;
@@ -266,8 +266,7 @@
   RawServoFontFaceRule* FindRuleForUserFontEntry(
       gfxUserFontEntry* aUserFontEntry);
 
-  nsresult StartLoad(gfxUserFontEntry* aUserFontEntry,
-                     const gfxFontFaceSrc* aFontFaceSrc);
+  nsresult StartLoad(gfxUserFontEntry* aUserFontEntry, uint32_t aSrcIndex);
   gfxFontSrcPrincipal* GetStandardFontLoadPrincipal();
   nsresult CheckFontLoad(const gfxFontFaceSrc* aFontFaceSrc,
                          gfxFontSrcPrincipal** aPrincipal, bool* aBypassCache);
@@ -277,8 +276,8 @@
   nsresult SyncLoadFontData(gfxUserFontEntry* aFontToLoad,
                             const gfxFontFaceSrc* aFontFaceSrc,
                             uint8_t*& aBuffer, uint32_t& aBufferLength);
-  nsresult LogMessage(gfxUserFontEntry* aUserFontEntry, const char* aMessage,
-                      uint32_t aFlags, nsresult aStatus);
+  nsresult LogMessage(gfxUserFontEntry* aUserFontEntry, uint32_t aSrcIndex,
+                      const char* aMessage, uint32_t aFlags, nsresult aStatus);
 
   void InsertRuleFontFace(FontFace* aFontFace, StyleOrigin aOrigin,
                           nsTArray<FontFaceRecord>& aOldRecords,